main.py

```diff
@@ -31,11 +43,221 @@ def setup(self):
+        """Method to start the process to enable table
+        First, send event notifying NApps about their
+        new table set up.
+        """
+        name = "enable_table"
+        self.emit_event(name, content)
+
+    def build_content(self, pipeline: dict) -> dict:
+        """Build content to be sent through an event"""
+        content = {}
+        for table in pipeline["multi_table"]:
+            table_id = table["table_id"]
+            for napp in table.get("napps_table_groups", {}):
+                if napp not in content:
+                    content[napp] = {}
+                for flow_type in table["napps_table_groups"][napp]:
+                    content[napp][flow_type] = table_id
+        return content
+
+    def emit_event(self, name: str, content: dict = None):
+        """Send event"""
+        context = "kytos/of_multi_table"
+        event_name = f"{context}.{name}"
+        event = KytosEvent(name=event_name, content=content)
+        self.controller.buffers.app.put(event)
+
+    @listen_to("kytos/(mef_eline|coloring|of_lldp).enable_table")
+    def on_enable_table(self, event):
+        """Listen for NApps responses"""
+        self.handle_enable_table(event)
+
+    def handle_enable_table(self, event):
+        """Handle NApps responses from enable_table
+        Second, wait for all the napps to respond"""
+        napp = event.name.split('/')[1].split('.')[0]
+        # Check against the last current table
+        self.required_napps.remove(napp)
+        if self.required_napps:
+            # There are more required napps, 'waiting' responses
+            return
+        self.get_flows_to_be_installed()
+
+    @retry(
+            stop=stop_after_attempt(3),
+            wait=wait_fixed(20),
+            before_sleep=before_sleep,
+            retry=retry_if_exception_type(RequestError))
+    def get_installed_flows(self) -> Optional[Dict]:
+        """Get flows from flow_manager"""
+        command = "v2/stored_flows?state=installed"
+        response = httpx.get(f"{FLOW_MANAGER_URL}/{command}", timeout=20)
+
+        if response.is_server_error:
+            raise RequestError(message=f"{response.text} on {command}")
+
+        return response.json()
+
+    def get_flows_to_be_installed(self):
+        """Get flows from flow manager so this NApp can modify them
+        Third, install the flows with different table_id"""
+        pipeline = self.pipeline_controller.get_active_pipeline()
+        if not pipeline or pipeline.get("status") == "enabled":
+            # Default or enabled pipeline, not need to get flows
+            return
+
+        pipeline_id = pipeline['id']
+        if "disabl" in pipeline.get("status", ""):
+            pipeline = self.default_pipeline
+
+        try:
+            flows_by_swich = self.get_installed_flows()
+        except tenacity.RetryError:
+            status = pipeline.get("status")
+            status = "enabling_error" if status else "disabling_error"
```

Comment:
Let's suppose a pipeline is currently in `disabling_error`, and then a user tries to make a request to  `POST /v1/pipeline/{pipeline_id}/disable`, which will set the pipeline status again as `disabling` (which is as expected), since it's trying to potentially fix it assuming it was a recoverable error like when retries fails here for instance (extremely rare but it can happen), in this case the prior state would be `disabling`, so `if status` would be true and it would result in `enabling_error`, which wouldn't be coherent, right? Did I follow this correctly? Can you double check?

Reply:
`pipeline` is replace with `self.default_pipeline` before when a status has the str `disabl` in its value, [line](https://github.com/kytos-ng/of_multi_table/blob/3a56caf0c94986bd6fc50fa5552c5eec27ba6798/main.py#L150-L151). `status` can only get `enabling` and `enabling_error` from the dictionary.

Reply:
Oh, I see, Aldo. Right. Let's close this thread then.